manageiq-operator/pkg/helpers/miq-components/httpd.go

```diff
@@ -81,7 +84,16 @@ func Ingress(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme) (*extensionsv1bet
 	return ingress, f
 }
 
-func HttpdConfigMap(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme) (*corev1.ConfigMap, controllerutil.MutateFn) {
+func HttpdConfigMap(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme) (*corev1.ConfigMap, controllerutil.MutateFn, error) {
+
+	if cr.Spec.HttpdAuthenticationType == "openid-connect" && cr.Spec.OIDCProviderURL != "" && cr.Spec.OIDCOAuthIntrospectionURL == "" {
+		introspectionURL, err := fetchIntrospectionUrl(cr.Spec.OIDCProviderURL)
+		if err != nil {
+			return nil, nil, err
+		}
+		(&cr.Spec).OIDCOAuthIntrospectionURL = introspectionURL
```

Comment:
You don't need the & and parens here. `cr.Spec.OIDCOAuthIntrospectionURL = introspectionURL` will do what you want as `cr` is already a pointer.

I assume you lifted this from [`Initialize`](https://github.com/ManageIQ/manageiq-pods/blob/26539c7eed15b4d07ff215e1416ee21be34cff1f/manageiq-operator/pkg/apis/manageiq/v1alpha1/manageiq_types.go#L290-L295)?

If that's the case that's different because we're using an intermediate local to store the spec so we don't need `m.Spec` everywhere.